feat: support local plugin file

LoliWolf opened the pull request:
feat: support local plugin file with config

LoliWolf committed:
feat: 支持本地文件

LoliWolf committed:
feat: new format

dogukancagatay commented on README.md:
````diff
@@ -1,66 +1,125 @@
+---
 
-For a Custom Plugin Repository, you will need plugin files, `updatePlugins.xml` file and serve these on a web server.
+```markdown
````
Why is the whole updated readme under this code block?

dogukancagatay commented on repo-builder.go:
```diff
@@ -116,285 +69,217 @@ type ReleaseDTO struct {
 		ServerUrl: "http://localhost:3000",
 		BindIp:    "0.0.0.0",
-		Port:      "3000",
+		Port:      "30300",
```
Why do you need to change the default port?

dogukancagatay commented on .DS_Store:
I believe this file is accidentally committed

LoliWolf committed:
Update README.md

LoliWolf committed:
Update README.md

LoliWolf committed:
chore: port

LoliWolf committed:
chore: remove file

LoliWolf committed:
fix(repo-builder): 优化 HTTP服务启动代码

- 在 http.ListenAndServe 调用中添加错误处理

LoliWolf committed:
Merge remote-tracking branch 'origin/master'

LoliWolf committed:
refactor(repo-builder): 重构代码以提高错误处理和日志记录

- 在 processPlugin 和 buildRepository 函数中添加错误处理逻辑- 使用 log.Fatalf 记录关键错误并退出程序
- 优化代码结构，提高可读性和可维护性

LoliWolf commented:
already changed
